### DISCUSSION TITLE
Can the output format of the DateTime field be specified during xml serialization?

### DISCUSSION THREAD

**Discussion by pedoc** [dotnet/runtime#86291] [Q&A]
For example, is the following code possible?

```csharp
public class A{
[DateTimeFormat("yyyy-MM-dd HH:mm:ss.fff")]
public DateTime Age{get;set;}
}
```

serialized result
```xml
<A>
<Age>2023-01-01 12:00:00.111</Age>
</A>
```

**Comment by am11** [comment]
You will need an additional field for string format projection. Something like:

```c#
using System;
using System.IO;
using System.Xml.Serialization;

XmlSerializer serializer = new(typeof(A));
string xmlString = @"
<A>
  <Age>2023-01-01 12:00:00.111</Age>
</A>
";

using TextReader reader = new StringReader(xmlString);
if (serializer.Deserialize(reader) is A a)
{
    Console.WriteLine("Parsed Age: {0}", a.AgeString);
}
else
{
    Console.WriteLine("Deserialization failed!");
}

public class A
{
    [XmlIgnore]
    public DateTime Age { get; private set; }
    
    [XmlElement(nameof(Age))]
    public string AgeString
    {
        get => Age.ToString("yyyy-MM-dd HH:mm:ss.fff");
        set => Age = DateTime.Parse(value);
    }
}
```

Output:
> Parsed Age: 2023-01-01 12:00:00.111

**Reply by pedoc** [reply]
Hi,thank you for your solution, I am currently doing the same. 
But I hope there is a more elegant solution to avoid redundant shadow field.

Redundant fields make the model bigger and confusing. It would be much better if you can customize the Converter or complete it through Attribute like JSON serialization.

I found several other xml serialization/deserialization implementations on github, but I'd still like to see if I can do this from the BCL library.
